File: app/src/main/java/org/lifefortheorc/tudounotepad/activity/ViewActivity.java
package org.lifefortheorc.tudounotepad.activity;

import android.app.ActionBar;
import android.app.Activity;
import android.content.Intent;
import android.os.Bundle;
import android.view.MenuItem;
import android.widget.TextView;

import org.lifefortheorc.tudounotepad.R;
import org.lifefortheorc.tudounotepad.model.NoteModel;

import butterknife.Bind;
import butterknife.ButterKnife;

/**
 * Created by geminiwen on 15/7/6.
 */
public class ViewActivity extends Activity {

    @Bind(R.id.tv_content)
    public TextView mTextViewContent;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_view);

    }

    @Override
    public void onContentChanged() {
        super.onContentChanged();

        ButterKnife.bind(this);

        ActionBar actionBar = getActionBar();
        actionBar.setDisplayHomeAsUpEnabled(true);

        mTextViewContent.setTransitionName("content");

        Intent intent = getIntent();
        long id = intent.getLongExtra("id", -1);

        if (id == -1) {
            this.finish();
            return;
        }

        NoteModel note = NoteModel.queryById(id);
        String content = note.getContent();
        mTextViewContent.setText(content);
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        int id = item.getItemId();
        switch (id) {
            case android.R.id.home: {
                this.finish();
                break;
            }
        }
        return super.onOptionsItemSelected(item);
    }
}

File: app/src/main/kotlin/org/lifefortheorc/tudounotepad/activity/ViewActivity.kt
package org.lifefortheorc.tudounotepad.activity

import android.os.Bundle
import android.view.MenuItem
import android.widget.TextView
import me.imid.swipebacklayout.lib.SwipeBackLayout
import me.imid.swipebacklayout.lib.app.SwipeBackActivity
import org.lifefortheorc.tudounotepad.R
import org.lifefortheorc.tudounotepad.base.BaseSwipeActivity
import org.lifefortheorc.tudounotepad.model.NoteModel

/**
 * Created by geminiwen on 15/7/10.
 */
public class ViewActivity(): BaseSwipeActivity() {
    var textViewContent: TextView? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_view)
    }

    override fun onContentChanged() {
        super.onContentChanged()

        val actionBar = getActionBar()
        actionBar.setDisplayHomeAsUpEnabled(true)

        textViewContent = findViewById(R.id.tv_content) as TextView

        textViewContent?.setTransitionName("content")

        val intent = getIntent()
        val id = intent.getLongExtra("id", -1L)

        if (id == -1L) {
            this.finish()
            return
        }

        val note = NoteModel.queryById(id)
        val content = note.content
        textViewContent?.setText(content)
    }

}
